### PR TITLE
feat: update dependencies to polkadot v1 release

### LINKED ISSUE #1148: upgrade to polkadot v1.0.0
**Is your feature request related to a problem? Please describe.**
Upgrading to the latest version is a proactive approach to ensure that our network stays up-to-date with the latest developments and advancements. 

**Describe the solution you'd like**
All the para chain dependencies need to be upgraded to the latest [Polkadot v1.0.0](https://github.com/paritytech/polkadot/releases/tag/v1.0.0) release 

**Additional context**
Update [Xcm-simulator](https://github.com/shaunxw/xcm-simulator) to the latest release as it is one of the dependencies.

### PR DESCRIPTION
**Pull request by nakul1010**
Closes #1148 

The Major changes in the code
- Adds serve feature flag to the primitives. This way, serde (de)serialization can also be used in no_std environments. ([PR 1307](https://github.com/paritytech/substrate/pull/13027))
- Remove node Cli Parameters (Done in[ 0.9.43 release](https://github.com/paritytech/polkadot/releases/tag/v0.9.43) )
- Update `GenesisConfig` (Helping PRs- [14108](https://github.com/paritytech/substrate/pull/14108/files))

For more info on the release, check out the [tag here](https://github.com/paritytech/polkadot/releases/tag/v1.0.0)

Release Analysis

- [Polkadot v0.9.43](https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-43/3158) 
- [Polkadot v1.0.0](https://forum.polkadot.network/t/polkadot-release-analysis-v1-0-0/3585)

### CODE REVIEW

**Comment by nakul1010** [comment]
blocked by ORML [issue](https://github.com/open-web3-stack/open-runtime-module-library/issues/940)

**Comment by nakul1010** [comment]
The PR is unblocked as ORML dependencies are updated to `1.0.0`.
Reference [PR](https://github.com/open-web3-stack/open-runtime-module-library/pull/942)

**Comment by nakul1010** [comment]
Blocked by [mocktopus crate] dependency (https://github.com/CodeSandwich/Mocktopus/issues/75)

File: parachain/runtime/interlay/src/lib.rs [hunk 1]
```diff
@@ -1446,7 +1445,7 @@ mod benches {
 
         // Other
         [cumulus_pallet_xcmp_queue, XcmpQueue]
-        [frame_system, frame_system_benchmarking::Pallet::<Runtime>]
+        // [frame_system, frame_system_benchmarking::Pallet::<Runtime>]
```
Comment (nakul1010) [inline parachain/runtime/interlay/src/lib.rs:1448]: Benchmarking fails for `frame_system`, need to intialize set code, something like [here](https://github.com/mangata-finance/cumulus/blob/7bcbff3605e38fb59d6882e30f7a8a10060234d0/parachain-template/runtime/src/lib.rs#L711) but it fails with 
`function or associated item not found in Pallet<Runtime>` even after feature gating through `runtime-benchmarks`

File: crates/security/src/lib.rs [hunk 2]
```diff
@@ -110,11 +110,14 @@ pub mod pallet {
         }
     }
 }
-
+use frame_system::pallet_prelude::BlockNumberFor;
```
Comment (gregdhill) [inline crates/security/src/lib.rs:113]: Move import to top

File: parachain/runtime/common/src/lib.rs [hunk 3]
```diff
@@ -96,16 +96,16 @@ where
 }
 
 pub struct AndBarrier<T: ShouldExecute, U: ShouldExecute>(PhantomData<(T, U)>);
-
+use xcm_executor::traits::Properties;
```
Comment (gregdhill) [inline parachain/runtime/common/src/lib.rs:99]: Move to top of file

File: parachain/runtime/interlay/src/lib.rs [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 #![cfg_attr(not(feature = "std"), no_std)]
 // `construct_runtime!` does a lot of recursion and requires us to increase the limit to 256.
-#![recursion_limit = "256"]
+#![recursion_limit = "512"]
```
Comment (gregdhill) [inline parachain/runtime/interlay/src/lib.rs:5]: Why did you need to increase this? Curious if you ran into a particular error

File: parachain/runtime/interlay/src/lib.rs [hunk 5]
```diff
@@ -322,6 +322,7 @@ impl pallet_aura::Config for Runtime {
     type AuthorityId = AuraId;
     type DisabledValidators = ();
     type MaxAuthorities = MaxAuthorities;
+    type AllowMultipleBlocksPerSlot = ConstBool<false>;
```
Comment (gregdhill) [inline parachain/runtime/interlay/src/lib.rs:325]: I am assuming this is related to the async backing feature? Any idea if / when we should set this to true?

File: parachain/runtime/interlay/src/lib.rs [hunk 6]
```diff
@@ -394,6 +395,7 @@ impl pallet_transaction_payment::Config for Runtime {
 impl pallet_sudo::Config for Runtime {
     type RuntimeCall = RuntimeCall;
     type RuntimeEvent = RuntimeEvent;
+    type WeightInfo = ();
```
Comment (gregdhill) [inline parachain/runtime/interlay/src/lib.rs:398]: Needs benchmarks, probably not a blocker can create a follow-up ticket

File: parachain/runtime/interlay/src/lib.rs [hunk 7]
```diff
@@ -1268,30 +1267,30 @@ construct_runtime! {
         CollatorSelection: collator_selection::{Pallet, Call, Storage, Event<T>, Config<T>} = 81,
         Session: pallet_session::{Pallet, Call, Storage, Event, Config<T>} = 82,
         Aura: pallet_aura::{Pallet, Storage, Config<T>} = 83,
-        AuraExt: cumulus_pallet_aura_ext::{Pallet, Storage, Config} = 84,
-        ParachainSystem: cumulus_pallet_parachain_system::{Pallet, Call, Config, Storage, Inherent, Event<T>} = 85,
-        ParachainInfo: parachain_info::{Pallet, Storage, Config} = 86,
+        AuraExt: cumulus_pallet_aura_ext::{Pallet, Storage, Config<T>} = 84,
+        ParachainSystem: cumulus_pallet_parachain_system::{Pallet, Call, Config<T>, Storage, Inherent, Event<T>,ValidateUnsigned} = 85,
```
Comment (gregdhill) [inline parachain/runtime/interlay/src/lib.rs:1271]: Why is `ValidateUnsigned` required?

File: parachain/runtime/interlay/src/lib.rs (hunk 1 above)
Comment (gregdhill) [reply parachain/runtime/interlay/src/lib.rs:1448]: Have you propagated the feature through `parachain-system`? https://github.com/paritytech/cumulus/blob/d4bb2215bb28ee05159c4c7df1b3435177b5bf4e/pallets/parachain-system/src/lib.rs#L1428-L1429

File: parachain/runtime/kintsugi/Cargo.toml [hunk 8]
```diff
@@ -32,7 +32,7 @@ sp-weights = { git = "https://github.com/paritytech/substrate", branch = "polkad
 frame-support = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.31", default-features = false }
 frame-executive = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.31", default-features = false }
 frame-system = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.31", default-features = false }
-frame-system-benchmarking = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.31", default-features = false, optional = true }
+frame-system-benchmarking = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.31", default-features = false , optional = true}
```
Comment (gregdhill) [inline parachain/runtime/kintsugi/Cargo.toml:35]: Revert formatting change

File: parachain/runtime/kintsugi/src/lib.rs [hunk 9]
```diff
@@ -1269,31 +1268,32 @@ construct_runtime! {
...
         UnknownTokens: orml_unknown_tokens::{Pallet, Storage, Event} = 95,
 
         // # Lending & AMM
-        Loans: loans::{Pallet, Call, Storage, Event<T>, Config} = 100,
+        Loans: loans::{Pallet, Call, Storage, Event<T>, Config<T>} = 100,
         DexGeneral: dex_general::{Pallet, Call, Storage, Event<T>} = 101,
         DexStable: dex_stable::{Pallet, Call, Storage, Event<T>}  = 102,
         DexSwapRouter: dex_swap_router::{Pallet, Call, Event<T>} = 103,
 
+
```
Comment (gregdhill) [inline parachain/runtime/kintsugi/src/lib.rs:1290]: Revert

File: parachain/runtime/runtime-tests/Cargo.toml [hunk 10]
```diff
@@ -21,75 +21,75 @@ flate2 = "1.0"
...
-pallet-scheduler = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.31" }
-pallet-multisig = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.31" }
-pallet-sudo = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.31" }
-pallet-preimage = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.31" }
-pallet-contracts = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.31", default-features = false }
-pallet-contracts-primitives = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.31", default-features = false }
-
-frame-system-rpc-runtime-api = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.31" }
-pallet-transaction-payment-rpc-runtime-api = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.31" }
+sp-std = { git = "https://github.com/paritytech/substrate", rev = "948fbd2fd1233dc26dbb9f9bbc1d2cca2c03945d" }
```
Comment (gregdhill) [inline parachain/runtime/runtime-tests/Cargo.toml:24]: Use polkadot-v1 branch for these

File: parachain/runtime/runtime-tests/src/parachain/contracts.rs [hunk 11]
```diff
@@ -63,7 +63,8 @@ fn test_basic_contract() {
             Code::Existing(blob_hash),
             input,
             vec![],
-            true,
+            DebugInfo::Skip,
+            pallet_contracts::CollectEvents::Skip,
```
Comment (gregdhill) [inline parachain/runtime/runtime-tests/src/parachain/contracts.rs:67]:
```suggestion
            CollectEvents::Skip,
```

File: parachain/runtime/runtime-tests/src/parachain/contracts.rs [hunk 12]
```diff
@@ -177,7 +181,8 @@ fn test_btc_swap_contract() {
                 GAS_LIMIT,
                 None,
                 execute_trade,
-                false,
+                DebugInfo::Skip,
+                pallet_contracts::CollectEvents::Skip,
```
Comment (gregdhill) [inline parachain/runtime/runtime-tests/src/parachain/contracts.rs:185]:
```suggestion
                CollectEvents::Skip,
```

File: parachain/runtime/runtime-tests/src/relaychain/mod.rs [hunk 13]
```diff
@@ -1,4 +1,4 @@
-mod kusama_cross_chain_transfer;
-pub mod kusama_test_net;
-mod polkadot_cross_chain_transfer;
-pub mod polkadot_test_net;
+// mod kusama_cross_chain_transfer;
+// pub mod kusama_test_net;
+// mod polkadot_cross_chain_transfer;
+// pub mod polkadot_test_net;
```
Comment (gregdhill) [inline parachain/runtime/runtime-tests/src/relaychain/mod.rs:4]: Revert, why did you delete these files?

File: parachain/src/eth.rs [hunk 14]
```diff
@@ -27,12 +27,8 @@ use std::{
 pub type FrontierBackend = fc_db::Backend<Block>;
 
 pub fn db_config_dir(config: &Configuration) -> PathBuf {
-    let application = &config.impl_name;
-    config
-        .base_path
-        .as_ref()
-        .map(|base_path| base_path.config_dir(config.chain_spec.id()))
-        .unwrap_or_else(|| BasePath::from_project("", "", application).config_dir(config.chain_spec.id()))
+    let _application = &config.impl_name;
```
Comment (gregdhill) [inline parachain/src/eth.rs:30]: Remove

File: parachain/src/service.rs [hunk 15]
```diff
@@ -486,11 +489,23 @@ where
     };
 
     if parachain_config.offchain_worker.enabled {
-        sc_service::build_offchain_workers(
```
Comment (gregdhill) [inline parachain/src/service.rs:489]: Was this removed upstream?

File: .github/workflows/cargo-test.yml [hunk 16]
```diff
@@ -47,15 +49,13 @@ jobs:
           ./target/release/interbtc-parachain benchmark pallet \
             --pallet '*' \
             --extrinsic '*' \
-            --execution=wasm \
```
[the remark this answers is not in the document]
Comment (nakul1010) [reply .github/workflows/cargo-test.yml:50]: Depreciated, [Helping PR](https://github.com/paritytech/substrate/pull/14387)

File: crates/annuity/src/mock.rs [hunk 17]
```diff
@@ -132,13 +123,21 @@ parameter_types! {
...
+    }
+}
 
 impl Config for Test {
     type AnnuityPalletId = AnnuityPalletId;
     type RuntimeEvent = RuntimeEvent;
     type Currency = Balances;
     type BlockRewardProvider = MockBlockRewardProvider;
-    type BlockNumberToBalance = Identity;
+    type BlockNumberToBalance = BlockNumberToBalance;
```
[the remark this answers is not in the document]
Comment (nakul1010) [reply crates/annuity/src/mock.rs:140]: had tried it already, the Conversion traits are not implemented for `Identity`  
`the trait `sp_runtime::traits::Convert<u128, u128>` is not implemented for `hidden_include::Identity`

File: parachain/runtime/interlay/src/lib.rs (hunk 4 above)
Comment (nakul1010) [reply parachain/runtime/interlay/src/lib.rs:5]: yaa,   
`error: recursion limit reached while expanding frame_support::tt_call!`

File: parachain/runtime/interlay/src/lib.rs (hunk 6 above)
Comment (nakul1010) [reply parachain/runtime/interlay/src/lib.rs:398]: issue #1185 will resolve it

File: parachain/runtime/interlay/src/lib.rs (hunk 5 above)
Comment (nakul1010) [reply parachain/runtime/interlay/src/lib.rs:325]: Described in an inline comment [here](https://github1s.com/paritytech/substrate/blob/HEAD/frame/aura/src/lib.rs#L285-L287). For us, it should be false based on [this comment](https://github1s.com/paritytech/substrate/blob/HEAD/frame/aura/src/lib.rs#L111-L114)

File: parachain/src/service.rs (hunk 15 above)
Comment (nakul1010) [reply parachain/src/service.rs:493]: yes

File: parachain/runtime/interlay/src/lib.rs (hunk 7 above)
Comment (nakul1010) [reply parachain/runtime/interlay/src/lib.rs]: removed

File: parachain/runtime/runtime-tests/src/relaychain/mod.rs (hunk 13 above)
Comment (nakul1010) [reply parachain/runtime/runtime-tests/src/relaychain/mod.rs]: While reverting, it found out that `decl_test_parachains` macro requires `palances_pallet` which is not present in interlay/kintsugi runtime.
[Macro definition](https://github1s.com/paritytech/cumulus/blob/polkadot-v1.0.0/xcm/xcm-emulator/src/lib.rs#L497)

Comment (nakul1010) [reply parachain/runtime/runtime-tests/src/relaychain/mod.rs]: Issue #1190 will resolve it.

File: crates/bitcoin/src/address.rs [hunk 18]
```diff
@@ -13,8 +14,10 @@ use secp256k1::{constants::PUBLIC_KEY_SIZE, Error as Secp256k1Error, PublicKey a
 /// A Bitcoin address is a serialized identifier that represents the destination for a payment.
 /// Address prefixes are used to indicate the network as well as the format. Since the Parachain
 /// follows SPV assumptions we do not need to know which network a payment is included in.
-#[derive(Encode, Decode, Clone, Ord, PartialOrd, PartialEq, Eq, Debug, Copy, TypeInfo, MaxEncodedLen)]
-#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize, std::hash::Hash))]
+#[derive(
```
Comment (sander2) [inline crates/bitcoin/src/address.rs:17]: Why do you unconditionally implement the serde traits? I realize that it's now possible, but we don't need it in the runtime so imo better not to bloat the wasm blob

File: crates/bitcoin/src/formatter.rs [hunk 19]
```diff
@@ -103,10 +103,10 @@ impl TryFormat for CompactUint {
     fn try_format<W: Writer>(&self, w: &mut W) -> Result<(), Error> {
         if self.value < 0xfd {
             (self.value as u8).try_format(w)?;
-        } else if self.value < u16::max_value() as u64 {
+        } else if self.value < u16::MAX as u64 {
```
Comment (sander2) [inline crates/bitcoin/src/formatter.rs:106]: just curious: was max_value deprecated or something?

**Review by sander2 — COMMENTED** [review]
I don't know if you're aware of these, but imo they're very helpful. Can you also put them in the pr info?

https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-43/3158
https://forum.polkadot.network/t/polkadot-release-analysis-v1-0-0/3585

**Comment by sander2** [comment]
And can you make a ticket to address https://github.com/paritytech/substrate/pull/13958 please

File: parachain/runtime/kintsugi/src/lib.rs [hunk 20]
```diff
@@ -2198,17 +2195,8 @@ impl_runtime_apis! {
             code: pallet_contracts_primitives::Code<Hash>,
             data: Vec<u8>,
             salt: Vec<u8>,
-        ) -> pallet_contracts_primitives::ContractInstantiateResult<AccountId, Balance> {
-            if !contracts::EnableContracts::get() {
```
[the remark this answers is not in the document]
Comment (nakul1010) [reply parachain/runtime/kintsugi/src/lib.rs:2202]: my bad, added the check back.

File: crates/bitcoin/src/address.rs (hunk 18 above)
Comment (nakul1010) [reply crates/bitcoin/src/address.rs]: by adding a feature gate to `Address` type, `interbtc_primitive` types need to be feature-gated for serde, which would indeed require a feature gate for `vault-registry` types. but then `frame_support` requires `Serialize`, & `Deserialize` and throws the error 
```

  error[E0277]: the trait bound `interbtc_primitives::CurrencyId: Serialize` is not satisfied
      --> /Users/nakul/Desktop/Interlay_Work/node/interbtc/crates/vault-registry/src/lib.rs:69:1
       |
  69   | #[frame_support::pallet]
       | ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Serialize` is not implemented for `interbtc_primitives::CurrencyId`
  ...
  648  |         pub minimum_collateral_vault: Vec<(CurrencyId<T>, BalanceOf<T>)>,
       |         ---------------------------------------------------------------- required by a bound introduced by this call
       |
       = help: the following other types implement trait `Serialize`:
                 &'a T
                 &'a mut T
                 ()
                 (T0, T1)
                 (T0, T1, T2)
                 (T0, T1, T2, T3)
                 (T0, T1, T2, T3, T4)
                 (T0, T1, T2, T3, T4, T5)
               and 205 others
       = note: required for `(interbtc_primitives::CurrencyId, <T as currency::Config>::Balance)` to implement `Serialize`
       = note: 1 redundant requirement hidden
       = note: required for `frame_benchmarking::Vec<(interbtc_primitives::CurrencyId, <T as currency::Config>::Balance)>` to implement `Serialize`
  note: required by a bound in `frame_support::serde::ser::SerializeStruct::serialize_field`
      --> /Users/nakul/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.186/src/ser/mod.rs:1871:12
       |
  1871 |         T: Serialize;
       |            ^^^^^^^^^ required by this bound in `SerializeStruct::serialize_field`


```

Comment (sander2) [reply crates/bitcoin/src/address.rs:17]: Why can't we leave the serialization as-is? Currently some types (notably, those that are part of the genesisconfig), but in many other places they are std-only

Comment (nakul1010) [reply crates/bitcoin/src/address.rs:17]: after a bit more digging found out, vault-registry's `GenesisConfig` use `interbtc-primitives` therefore they need to have  Serialize, & Deserialize traits implemented, 
but `vault-registry` depends on `interbtc-primitives` which further depends upon `bitcoin` crate. So [following feature flag](https://github.com/paritytech/substrate/pull/13027/files#diff-9315604b21af73f4d96659db77ab7ccae3b0c90e5bd6e76c442de3696b6ee41cR37-R42) needs to be added in `bitcoin` and `interbtc-primitives` crate 

Changes in `interbtc-primitives` required
- Add the mentioned  [serde feature](https://github.com/paritytech/substrate/pull/13027/files#diff-9315604b21af73f4d96659db77ab7ccae3b0c90e5bd6e76c442de3696b6ee41cR37-R42)  
- All `Serialize`, & `Deserialize` traits should be unable when `std` or `serde` feature is on. 

Changes in `bitcoin` required
- Add the mentioned [serde feature](https://github.com/paritytech/substrate/pull/13027/files#diff-9315604b21af73f4d96659db77ab7ccae3b0c90e5bd6e76c442de3696b6ee41cR37-R42) 
- All `Serialize`, & `Deserialize` traits should be unable when `std` or `serde` feature is on. 

Changes in `vault-registry` required
- unable the feature for primitives
`primitives = { package = "interbtc-primitives", path = "../../primitives", default-features = false, features = ["serde"] } `

I don't know if this is the solution u intend me to build, as it still adds the traits in `runtime`, regarding wasm blob do u want me to check the size increase if the traits are implemented by default and then move ahead?

**Comment by nakul1010** [comment]
> And can you make a ticket to address [paritytech/substrate#13958](https://github.com/paritytech/substrate/pull/13958) please

Yes i can, but currently i see we are taking the `proof_size` from `cumulus_primitives` [here](https://github.com/interlay/interbtc/blob/HEAD/parachain/runtime/interlay/src/lib.rs#L154-L155), should it be calculated based on interlay parachain block fullness?

**Review by gregdhill — APPROVED** [review]
